core/collector.class.inc.php

```diff
@@ -952,19 +977,14 @@ protected function FindAttr($sAttCode, $aExpectedAttrDef)
 	protected function DataSourcesAreEquivalent($aDS1, $aDS2)
 	{
 		foreach ($aDS1 as $sKey => $value) {
-			switch ($sKey) {
-				case 'friendlyname':
-				case 'user_id_friendlyname':
-				case 'user_id_finalclass_recall':
-				case 'notify_contact_id_friendlyname':
-				case 'notify_contact_id_finalclass_recall':
-				case 'notify_contact_id_obsolescence_flag':
-				case 'notify_contact_id_archive_flag':
-					// Ignore all read-only attributes
-					break;
+			if (in_array($sKey, Collector::READONLY_FIELDS)){
+				// Ignore all read-only attributes
+				continue;
+			}
 
+			switch ($sKey) {
 				case 'attribute_list':
-					foreach ($value as $sKey => $aDef) {
+					foreach ($value as $sKey2 => $aDef) {
```

Comment:
Since it is unused, better do this
```suggestion
					foreach ($value as $aDef) {
```